plugins/workload-management/src/test/java/org/opensearch/plugin/wlm/service/QueryGroupPersistenceServiceTests.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.service;
+
+import org.opensearch.cluster.ClusterName;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.search.ResourceType;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.opensearch.cluster.metadata.QueryGroup.builder;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.MEMORY_STRING;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.MONITOR_STRING;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.NAME_NONE_EXISTED;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.NAME_ONE;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils._ID_ONE;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils._ID_TWO;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.clusterSettingsSet;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.compareQueryGroups;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.preparePersistenceServiceSetup;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.queryGroupList;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.queryGroupOne;
+import static org.opensearch.plugin.wlm.QueryGroupTestUtils.queryGroupTwo;
+import static org.opensearch.plugin.wlm.service.QueryGroupPersistenceService.QUERY_GROUP_COUNT_SETTING_NAME;
+import static org.mockito.Mockito.mock;
+
+public class QueryGroupPersistenceServiceTests extends OpenSearchTestCase {
+
+    public void testCreateQueryGroup() {
```

Comment:
Add 1-2 line before each test case to make it easier for developers to understand the purpose of every test